src/components/kubectl/compatibility.ts

```diff
@@ -0,0 +1,65 @@
+import { ShellResult } from '../../shell';
+
+interface CompatibilityGuaranteed {
+    readonly guaranteed: true;
+}
+
+interface CompatibilityNotGuaranteed {
+    readonly guaranteed: false;
+    readonly didCheck: boolean;
+    readonly clientVersion: string;
+    readonly serverVersion: string;
+}
+
+export type Compatibility = CompatibilityGuaranteed | CompatibilityNotGuaranteed;
+
+export function isGuaranteedCompatible(c: Compatibility): c is CompatibilityGuaranteed {
+    return c.guaranteed;
+}
+
+export async function check(kubectlInvokeAsync: (cmd: string) => Promise<ShellResult>): Promise<Compatibility> {
+    const sr = await kubectlInvokeAsync('version --short');
+    if (sr.code !== 0) {
+        return {
+            guaranteed: false,
+            didCheck: false,
+            clientVersion: '',
+            serverVersion: ''
+        };
+    }
+
+    const lines = sr.stdout.split('\n').map((l) => l.trim()).filter((l) => l.length > 0);
```

Comment:
you can use the json output from `kubectl version -o json`, if that makes life any easier here…

Reply:
TIL! Thank you!